src/main/java/org/folio/rest/impl/TenantRefAPI.java

```diff
@@ -1,48 +1,41 @@
 package org.folio.rest.impl;
+
 import java.util.Map;
-import javax.ws.rs.core.Response;
-import org.folio.rest.jaxrs.model.TenantAttributes;
-import org.folio.rest.tools.utils.TenantLoading;
 
-import io.vertx.core.AsyncResult;
 import io.vertx.core.Context;
-import io.vertx.core.Handler;
-import io.vertx.core.Vertx;
-import io.vertx.core.logging.Logger;
-import io.vertx.core.logging.LoggerFactory;
+import io.vertx.core.Future;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.rest.annotations.Validate;
+import org.folio.rest.jaxrs.model.TenantAttributes;
+import org.folio.rest.tools.utils.TenantLoading;
 
 public class TenantRefAPI extends TenantAPI {
- private static final Logger log = LoggerFactory.getLogger(TenantRefAPI.class);
 
+  private static final String SAMPLE_LEAD = "sample-data";
+  private static final String SAMPLE_KEY = "loadSample";
+  private static final String REFERENCE_KEY = "loadReference";
+  private static final String REFERENCE_LEAD = "ref-data";
+
+  private static final Logger log = LogManager.getLogger();
+
+  @Validate
   @Override
-  public void postTenant(TenantAttributes ta, Map<String, String> headers,
-    Handler<AsyncResult<Response>> hndlr, Context cntxt) {
-    log.info("postTenant" );
-    Vertx vertx = cntxt.owner();
-    super.postTenant(ta, headers, res -> {
-      if (res.failed()) {
-        hndlr.handle(res);
-        return;
-      }
-      TenantLoading tl = new TenantLoading();
-      tl.withKey("loadReference").withLead("ref-data")
-        .withIdContent()
-        .add("groups")
-        .withIdContent()
-        .add("addresstypes")
-        .withKey("loadSample").withLead("sample-data")
-        .withIdContent()
-        .add("users")
-        .perform(ta, headers, vertx, res1 -> {
-          if (res1.failed()) {
-            hndlr.handle(io.vertx.core.Future.succeededFuture(PostTenantResponse
-              .respond500WithTextPlain(res1.cause().getLocalizedMessage())));
-            return;
-          }
-          hndlr.handle(io.vertx.core.Future.succeededFuture(PostTenantResponse
-            .respond201WithApplicationJson("")));
+  Future<Integer> loadData(TenantAttributes attributes, String tenantId,
+                           Map<String, String> headers, Context vertxContext) {
+    return super.loadData(attributes, tenantId, headers, vertxContext)
+        .compose(superRecordsLoaded -> {
+          log.info("loading data to tenant");
+          TenantLoading tl = new TenantLoading();
+          tl.withKey(REFERENCE_KEY).withLead(REFERENCE_LEAD);
+          tl.withIdContent();
+          tl.add("groups");
+          tl.withIdContent();
+          tl.add("addresstypes");
+          tl.withKey(SAMPLE_KEY).withLead(SAMPLE_LEAD);
+          tl.withIdContent();
+          tl.add("users");
+          return tl.perform(attributes, headers, vertxContext, superRecordsLoaded);
```

Comment:
Personally, I find it easier to follow code when it spaced out a bit. In particular for methods that have sufficient parameters to have a line continuation, I think a blank line between the signature and the implementation helps to identify them easier.

```suggestion

    return super.loadData(attributes, tenantId, headers, vertxContext)
        .compose(superRecordsLoaded -> {
          log.info("loading data to tenant");
          
          TenantLoading tl = new TenantLoading();
          
          tl.withKey(REFERENCE_KEY).withLead(REFERENCE_LEAD);
          tl.withIdContent();
          tl.add("groups");
          tl.withIdContent();
          tl.add("addresstypes");
          
          tl.withKey(SAMPLE_KEY).withLead(SAMPLE_LEAD);
          tl.withIdContent();
          tl.add("users");
          
          return tl.perform(attributes, headers, vertxContext, superRecordsLoaded);
```